vmdb/app/models/ext_management_system.rb

```diff
@@ -1,20 +1,26 @@
 class ExtManagementSystem < ActiveRecord::Base
+  SUBCLASSES = %w(
+    EmsInfra
+    EmsCloud
+    EmsContainer
+  )
+
   include EmsRefresh::Manager
 
   def self.types
-    [EmsInfra, EmsCloud].collect(&:types).flatten
+    leaf_subclasses.collect(&:ems_type)
   end
 
   def self.supported_types
-    [EmsInfra, EmsCloud].collect(&:supported_types).flatten
+    supported_subclasses.collect(&:ems_type)
```

Comment:
I see, you changed from calling types/supported_types which returns an array... to just using the ems_type column.

Thanks, I missed that.